src/Features/Core/Portable/CodeRefactorings/AbstractRefactoringHelpersService.cs

```diff
@@ -336,13 +343,26 @@ protected virtual SyntaxNode DefaultNodeExtractor(SyntaxNode node, ISyntaxFactsS
                 return rightSide;
             }
 
+            // VB's arguments can have identifiers nested in ModifiedArgument -> we want
+            // indentifiers to represent parent node -> need to extract.
+            if (syntaxFacts.IsParametersIdentifier(node))
+            {
+                return node.Parent;
+            }
+
             if (extractParentsOfHeader)
             {
-                // Header: [TestAttribute] `public int a` { get; set; }
+                // Header: [Test] `public int a` { get; set; }
                 if (syntaxFacts.IsInPropertyDeclarationHeader(node))
                 {
                     return syntaxFacts.GetContainingPropertyDeclaration(node);
                 }
+
+                // Header: public C([Test]`int a` = 42) {}
```

Comment:
note: for a *parameter*  i would consider everything but hte attribute to be the header.  The reason for this is that, unlike a field/property, a paramter can only really have a trivial initializer.  So someone can be in that initializer and will still feel like they're "on the parameter" not "inside the initializer".  

That said, i don't care tons about this. We can start here and refine if ever necessary.

Reply:
IIRC there were tests guarding against that so I went with original behaviour. Thinking about it more, you're right, will update.